SentinelOne/cloud_funnel2.0/ingest/parser.yml

```
@@ -0,0 +1,736 @@
+name: cloudfunnel2.0
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: '{{original.message}}'
+
+  # Time conversions
+  - name: convert_timestamp
+    filter: '{{json_event.message.timestamp != null}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.timestamp}}'
+        output_field: timestamp
+
+  - name: convert_ossrc_process_start
+    filter: '{{json_event.message.get("osSrc.process.startTime") != None}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.get("osSrc.process.startTime")}}'
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_process_start
+    filter: '{{json_event.message.get("src.process.startTime") != None}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.get("src.process.startTime")}}'
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_parent_process_start
+    filter: '{{json_event.message.get("src.process.parent.startTime") != None}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.get("src.process.parent.startTime")}}'
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_target_process_start
+    filter: '{{json_event.message.get("tgt.process.startTime") != None}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.get("tgt.process.startTime")}}'
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_file_creation_time
+    filter: '{{json_event.message.get("tgt.file.creationTime") != None}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.get("tgt.file.creationTime")}}'
+        output_field: start_date
+        format: timestamp
+
+  - name: convert_file_modification_time
+    filter: '{{json_event.message.get("tgt.file.modificationTime") != None}}'
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.get("tgt.file.modificationTime")}}'
+        output_field: start_date
+        format: timestamp
+
+  # File extractions
+  - name: find_dll_name
+    filter: '{{json_event.message.get("event.category") == "module"}}'
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{json_event.message.get("module.path")}}'
+        output_field: file
+        pattern: '%{GREEDYDATA}[\\\/]+%{GREEDYDATA:name}$'
+
+  - name: find_reg_key_name_from_path
+    filter: '{{json_event.message.get("registry.keyPath") != None}}'
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{json_event.message.get("registry.keyPath")}}'
+        output_field: key
+        pattern: '%{GREEDYDATA}[\\\/]+%{GREEDYDATA:name}$'
```

I strongly suggest extracting the key (at least, used by rules) and the hive from the registry path.
```suggestion
        pattern: '%{DATA:hive}[\\]{2}%{GREEDYDATA:key}[\\\/]+%{GREEDYDATA:name}$'
```